src/topology/homotopy/contractible.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+
+import topology.homotopy.path
+import topology.homotopy.equiv
+
+/-!
+# Contractible spaces
+
+In this file, we define `contractible_space`, a space that is homotopy equivalent to `unit`.
+-/
+
+noncomputable theory
+
+
+namespace continuous_map
+
+/-- A map is nullhomotopic if it is homotopic to some constant map -/
+def nullhomotopic {X Y : Type*} [topological_space X] [topological_space Y] (f : C(X, Y)) : Prop :=
+∃ y : Y, homotopic f (continuous_map.const _ y)
+
+end continuous_map
+
+open continuous_map
+open_locale continuous_map
+
+/-- A contractible space is one that is homotopy equivalent to `unit`. -/
+class contractible_space (X : Type*) [topological_space X] : Prop :=
+(hequiv_unit [] : nonempty (X ≃ₕ unit))
+
+
+variables (X : Type*) [topological_space X] [contractible_space X]
+
+lemma id_nullhomotopic : (continuous_map.id X).nullhomotopic :=
```

Comment:
This should be called `contractible_space.id_nullhomotopic`, or something like that. You can maybe work in the `contractible_space` namespace.

Reply:
Oddly enough the opposite change was suggested here: https://github.com/leanprover-community/mathlib/pull/12731#discussion_r830260252

Reply:
:D Let's follow Johan's opinion, he's more experienced than me.